dali/operators/reader/loader/video/frames_decoder.h

```diff
@@ -109,18 +109,19 @@ class DLL_PUBLIC FramesDecoder {
    * 
    * @param memory_file Pointer to memory with video file data.
    * @param memory_file_size Size of memory_file in bytes.
+   * @param build_index If set to false index will not be build and some features are unavailible.
    * 
    * @note This constructor assumes that the `memory_file` and
    * `memory_file_size` arguments cover the entire video file, including the header.
    */
-  FramesDecoder(const char *memory_file, int memory_file_size);
+  FramesDecoder(const char *memory_file, int memory_file_size, bool build_index = true);
 
   /**
    * @brief Number of frames in the video
    * 
    */
   int64_t NumFrames() const {
-    return index_.size();
+    return av_state_->ctx_->streams[av_state_->stream_id_]->nb_frames;
```

Comment:
Is this value available for all containers or only with the index available (like avi)?

Reply:
According to the comment:
```
    int64_t nb_frames;                 ///< number of frames in this stream if known or 0
```
The code needs to be able to handle the situation where it is 0.

Reply:
I added a hard error to throw if there is no index and there is no `nb_frames`. In the things we support now, it should not happen. If it does, we will take it from there.